appcli/variables_manager.py

```diff
@@ -59,32 +64,30 @@ def get_all_variables(self):
         return self.__get_configuration()
 
     def set_variable(self, path: str, value: Union[str, bool, int, float]):
-        """Sets a value in the configuration.
+        """Sets a value in the configuration
 
         Type of value is not enforced but this might not serialise into yml in a deserialisable format.
 
         Args:
-            path (str): Dot notation for the setting. E.g. insilico.external.database.host
+            path (str): Dot notation for the setting. E.g. settings.insilico.external.database.host
             value: value for the setting
+
+        TODO: warn user if attempting to save variable not in main config.
         """
         configuration = self.__get_configuration()
-
-        path_elements = path.split(".")
-        parent_path = path_elements[:-1]
-
-        # ensure parent path exists
-        def ensure_path(parent, child):
-            if child not in parent:
-                parent[child] = {}
-            return parent[child]
-
-        reduce(ensure_path, parent_path, configuration)
-
-        # set the value
-        parent_element = reduce(lambda e, k: e[k], parent_path, configuration)
-        parent_element[path_elements[-1]] = value
-
-        self.__save(configuration)
+        path_elements = path.split(".")  # Divide path into array.
+        # Iterate through each context on path, to create a recursive dictionary.
+        # Main is a pointer to top dictionary object.
+        # Sub is a pointer to the current dictionary layer.
+        dictionary_main = dictionary_sub = {}
+        for field in path_elements:
+            # Apply value to bottom dictionary only.
+            dictionary_sub[field] = value if field is path_elements[-1] else {}
+            dictionary_sub = dictionary_sub[
+                field
+            ]  # Move sub pointer to next layer down.
+
+        self.__save(configuration | dictionary_main)
```

Comment:
With the original code it could store a value in a non existent field, but if the parent also didn't exist it would throw a NoneType exception.
This code allows it to create a field at any depth where the parent(s) don't exist.